pkg/kubelet/server/remotecommand/httpstream.go

```diff
@@ -167,15 +193,69 @@ func createStreams(req *http.Request, w http.ResponseWriter, supportedStreamProt
 
 	ctx.conn = conn
 	ctx.tty = opts.tty
+
 	return ctx, true
 }
 
 type protocolHandler interface {
 	// waitForStreams waits for the expected streams or a timeout, returning a
 	// remoteCommandContext if all the streams were received, or an error if not.
 	waitForStreams(streams <-chan streamAndReply, expectedStreams int, expired <-chan time.Time) (*context, error)
+	// supportsTerminalResizing returns true if the protocol handler supports terminal resizing
+	supportsTerminalResizing() bool
+}
+
+// v3ProtocolHandler implements the V3 protocol version for streaming command execution.
+type v3ProtocolHandler struct{}
```

Comment:
v1 was the original (unversioned) implementation that has bugs.

v2 fixed those bugs.

v3 adds support for terminal resizing.

I'd like to find a way to reuse code between v2 and v3 and beyond, if possible, but so far I haven't. Also, assuming we deprecate this and replace it with an HTTP/2 implementation, it will likely be similar to the websocket approach where we mux the streams/channels manually, so we won't have all these various, similar implementations (or we shouldn't, at least).